rfd/0056-buildbox-image-handling.md

```diff
@@ -0,0 +1,171 @@
+---
+authors: Trent Clarke (trent@goteleport.com)
+state: draft
+---
+
+# RFD 45 - Buidlbox image Handling 
+
+## What
+
+A "buildbox" image contains the tools environment for constructing executable 
+software from source. 
+
+## Why
+
+## Details
+
+### Goals
+
+Any workable solution must satisfy the following goals 
+
+ 1. The build system must be able to create repeatable builds, from source, of
+    any revision of the Teleport source at any time. 
+
+ 2. The same process (and its resulting images) should be used by all automated 
+    builds (e.g. CI, test & Release)
+
+ 3. Changes to the buildbox image must be reviewable and auditable.
+
+ 4. Changes to the buildbox should be opt-in, in the sense that changes to a 
+    buildbox should not be *forced* on a consumer - the consumer should deliberately 
+    reference the updated buildbox image.
+
+ 5. Using the buildbox images (at least in public-facing CI) must *not* require 
+    the use of privileged containers (e.g. require Docker-In-Docker), as this 
+    is a security threat.
+
+ 6. Creating an updated buildbox image once changes have been approved should 
+    be automatic (or at least a single push-button operation)
+
+ 7. Where multiple build environments are required, they should all be 
+    produced by the same process.
+
+### The current process
+
+Currently the buildbox images are constructed by Drone in response to a commits
```

Comment:
A year later, and **_THIS IS NO LONGER 100% TRUE_**.

- We now segregate the buildboxes based on the version of teleport they support (`teleport13`, `teleport12`, etc). This gets around the issue of changes to one major release polluting another release's buildbox, but doesn't really do anything for dealing with securing the supply chain. 
- The home repository for buildbox images is now `ECR`, so tag immutability is an option.